metaflow/cli.py

```diff
@@ -226,6 +226,12 @@ def output_dot(obj):
 def dump(obj, input_path, private=None, max_value_size=None, include=None, file=None):
 
     output = {}
+    if file is not None:
+        echo(
```

Comment:
How about throwing an error instead when both `file` and `max_value_size` are specified?

Reply:
yea that's a good callout -- let me rework this to `raise` an error instead

Reply:
I'm not a `click` expert but if you *don't* specify `--max-value-size` on the command line, would it pass the default value 1000 to `dump()` or None?

Reply:
i just did a simple test with this click file:

```
import click

@click.command()
@click.option('--n', default=5, show_default=True)
def dots(n=None):
    print(f"n: {n}")
    click.echo('.' * n)

if __name__ == "__main__":
    dots()
```

and then running -> `python test.py` yielded:

```
n: 5
.....
```

so that seems to imply that if the `--max-value-size` isn't passed it still populates the default value of 1000 to the `dump()` argument instead of `None`, which i guess begs the question of should we try to change the `max_value_size` if a `--file` argument is passed or should it error out

Reply:
I think at the end of the day, from the end user POV we want `--max-value-size` to default to unlimited if `--file` is specified, and to 1000 if not.

Reply:
Sure that makes sense - let me work that up and push up another pass at this